[design] 유저 정보 화면 UI

meanjoo opened the pull request:
## #️⃣연관된 이슈
- resolved: #28 

## 📝작업 내용 및 코드
- 유저 정보 화면 UI를 구현했습니다.
- 유저 정보 화면의 메뉴 부분을 따로 컴포저블 함수로 만들었습니다. 메뉴의 카테고리 이름, 하위 메뉴의 아이콘, 메뉴 이름, 클릭 이벤트만 넘기면 메뉴가 그려집니다.
  <details>
  <summary>예시 보기</summary>
  
  ```kotlin
  UserInfoMenus(
      title = "Pick",
      menus = listOf(
          MenuItem(
              imageVector = Icons.Outlined.Archive,
              contentDescription = "픽 보관함 메뉴 아이콘",
              menuTitle = "픽 보관함",
              onMenuClick = {}
          ),
          MenuItem(
              imageVector = Icons.Default.MusicNote,
              contentDescription = "등록한 픽 메뉴 아이콘",
              menuTitle = "등록한 픽",
              onMenuClick = {}
          )
      )
  )
  ```
  
  ![image (5)](https://github.com/user-attachments/assets/04245a9e-4582-47ce-b821-8caa1e3365f2)
  
  </details>

<details>
<summary>변경 전 디자인</summary>

<img src="https://github.com/user-attachments/assets/7ef8cddd-d799-437d-a42a-91aac64f24bb" width="300">

<img src="https://github.com/user-attachments/assets/68723a1c-0dcd-45f4-80a0-b035e2610e08" width="300">

</details>

<img src="https://github.com/user-attachments/assets/e606f942-5446-4c64-a77f-6767e7fef9f2" width="300">


## 💬리뷰 요구사항
- ~어라..... 픽 담기 구현했던 브랜치가 머지 안 된 상태인데 담기 했던 곳에서 새 브랜치 만들고 했더니 커밋 이력이 담기 부분도 다 나오네요 ㅠㅠ;; 죄송합니다.. `[design] 유저 정보 화면 디자인` 커밋만 봐주시면 될 것 같아요!! 아 이 커밋 안에서도 변경된 파일이 많은데 userinfo 패키지만 봐주시면 될 것 같아요. 나머지 변경 사항들은 공통 부분 옮긴 건데 커밋을 안 나눴네용..~

- 피그마에 있는 사용자 기본 프로필 이미지가 어디 있는지 몰라서 제가 아무거나 하나 찾아서 넣어뒀는데 피그마에 있는 이미지 갖고 계신 분 저에게 주시면 감사하겠습니다,,

- 메뉴의 이름이 카테고리는 영어이고 하위 메뉴는 한글이라 한글로 통일하고 싶은데 좋은 이름 없을까요? 제가 생각해본 건 [픽] 픽 보관함, 등록한 픽 / [설정] 닉네임, 알림 이렇게 나누는 겁니다! 아니면 어차피 알림을 못 할 것 같으니 [설정] 닉네임 이렇게만 있어도 괜찮을 것 같긴 해요. 저만 영어랑 한글이 혼용돼있는 게 별로라면 넘어가는 걸로 하죠 ㅎㅎ
  => 기존처럼 카테고리 이름은 영어, 하위 메뉴는 한글로 유지. 다만 [Settings] 아래에 설정이 있는 게 이상(해석하면 설정 하위에 설정)하여 [Settings] 하위 메뉴로 프로필, 알림으로 구분하기로 결정

meanjoo committed:
[design] 유저 정보 화면 디자인

meanjoo committed:
[design] 메뉴 부분 디자인 변경

yuni-ju commented:
현재 ripple 영역이 메뉴의 divider보다 더 튀어나와있는 것 같은데 이 부분도 padding이 함께 적용되면 좋을 것 같습니다! 

현재 클릭 시 영역 
<img width="200" src="https://github.com/user-attachments/assets/98ac3727-420d-4c2b-8c9d-feaaefa4f319">

meanjoo committed:
Merge branch 'develop' into design/#28-user-info

miller198 commented on app/src/main/java/com/squirtles/musicroad/main/navigations/MainNavigationActions.kt:
```diff
@@ -24,6 +24,16 @@ class MainNavigationActions(navController: NavHostController) {
+            restoreState = true
+        }
+    }
+
```
navigateToUserInfo 여기도 수정해주세요!

miller198 commented on app/src/main/java/com/squirtles/musicroad/common/Spacer.kt:
```diff
@@ -0,0 +1,14 @@
+fun VerticalSpacer(height: Int) = Spacer(Modifier.height(height.dp))
+
+@Composable
+fun HorizontalSpacer(width: Int) = Spacer(Modifier.width(width.dp))
```
Spacer를 쓰는 다른 코드들도 모두 대체되면 분리한 의미가 더 있을것같습니다!

meanjoo commented:
@yuni-ju 앗 디자인 변경하면서 생각하지 못했던 부분인데 반영하도록 하겠습니다!!

<img src="https://github.com/user-attachments/assets/37edc613-0af2-4218-873e-8357d2fe2bb6" width="300">

반영했습니당

miller198 commented on app/src/main/java/com/squirtles/musicroad/userinfo/TitleAndMenus.kt:
```diff
@@ -0,0 +1,104 @@
+import com.squirtles.musicroad.ui.theme.White
+
+@Composable
+internal fun TitleAndMenus(
```
함수 이름이 불명확해서 UserInfoMenus 같이 정확히 어떤 역할을 하는지 수정해주세요

meanjoo replied:
좋네요 UserInfoMenus로 변경하겠습니다!

meanjoo committed:
[refactor] ripple 영역 조정

miller198 reviewed (commented):
고생하셧씁니다!!!

meanjoo committed:
[refactor] 리뷰 반영

함수명 및 파일명 더 명확하게 수정: TitleAndMenus -> UserInfoMenus
Spacer 대신 공통 Spacer 컴포넌트 사용
UserInfo 내비게이션의 popUpTo 제거

meanjoo committed:
[design] 설정 하위 메뉴 변경
